src/views/collection-view/components/collection-meta-progress/helpers/parse-collection-progress.ts

```diff
@@ -0,0 +1,145 @@
+import { Collection, TutorialLite } from 'lib/learn-client/types'
+import { ApiCollectionTutorialProgress } from 'lib/learn-client/api/api-types'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+
+/**
+ * Given progress data for a particular collection,
+ * and a count of all tutorials in that collection,
+ * Return booleans representing collection progress state.
+ */
+function parseCollectionProgress(
```

Comment:
This helper handles the logic to get all the little bits we need to know exactly what to render in `CollectionMetaProgress`.